src/masternodes/mn_checks.cpp

```diff
@@ -1111,8 +1112,13 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor {
                 if (!minted)
                     return minted;
 
-                CalculateOwnerRewards(*mintable.val);
-                auto res = mnview.AddBalance(*mintable.val, CTokenAmount{tokenId, amount});
+                CScript mintTo{*mintable.val};
+                if (height >= grandCentralNextHeight && !obj.to.empty()) {
```

Comment:
Need to reject a populated to address below the fork height. The problem is that new clients will accept the TX with the to address below fork height and old clients will not be able to deserialise the TX as it will not have the to field. For compatibility we want new clients to reject use of the to field below the fork height, not just ignore it. After the fork such a check can be removed when all users are on the new version.

Also, let's check that to address is valid, extract the destination using ExtractDestination(), if that fails then reject the TX, after call IsValidDestination() on the destination and if that fails reject the TX. We have several places were we just trust whatever the script address is, but we really should be checking that addresses are valid and that users are not sending to junk destinations.

Reply:
Fixed: https://github.com/DeFiCh/ain/pull/1650/commits/7f09763b10e6fa022abfe4f676bcd493eab6bb7f